src/main/java/org/concordion/internal/command/executeCommand/ExecuteCommand.java

```diff
@@ -17,6 +21,98 @@
     public ExecuteCommand() {
     }
 
+    @Override
+    public void modifyCommandCallTree(CommandCall commandCall, List<CommandCall> examples, List<CommandCall> beforeExamples) {
+        if (isTableElement(commandCall)) {
+            performTableModification(commandCall);
+        }
+
+        if (isListElement(commandCall)) {
+            performListModification(commandCall);
+        }
+    }
+
+    private void performListModification(CommandCall commandCall) {
+    }
+
+
+    private Map<Integer, CommandCall> populateCommandCallByColumnMap(Table table, CommandCall tableCommandCall) {
+        Map<Integer, CommandCall> commandCallByColumn = new HashMap<Integer, CommandCall>();
+
+        Row headerRow = table.getLastHeaderRow();
+        for (CommandCall childCall : tableCommandCall.getChildren().asCollection()) {
+            int columnIndex = headerRow.getIndexOfCell(childCall.getElement());
+            if (columnIndex == -1) {
+                throw new RuntimeException("Commands must be placed on <th> elements when using 'execute' or 'verifyRows' commands on a <table>.");
+            }
+            commandCallByColumn.put(columnIndex, childCall);
+        }
+
+        return commandCallByColumn;
+    }
+
+    private void performTableModification(CommandCall commandCall) {
+
+
+        /*
+
+        We have to:
+        * Copy the execute commandCall to each TR except for the header rows
+        * Copy the TH commandCall to each TD
+        * remove the table execute commandCall from the table
+        * remove the TH commandCalls from the TH rows.
+         */
+
+        Table table = new Table(commandCall.getElement());
+        Map<Integer, CommandCall> headerCommands = populateCommandCallByColumnMap(table, commandCall);
+
+        // copy the execute to each detail row.
+        Row[] detailRows = table.getDetailRows();
+        for (int i = 0; i < detailRows.length; i++) {
+            Row row = detailRows[i];
+
+            if (row.getCells().length != table.getLastHeaderRow().getCells().length) {
+                throw new RuntimeException("The <table> 'execute' command only supports rows with an equal number of columns. Detail row " + (i + 1) + " has a different number of columns to the last header row");
+            }
+
+            CommandCall rowCommand = duplicateCommandForDifferentElement(commandCall, row.getElement());
+            rowCommand.transferToParent(commandCall);
+            Element[] cells = row.getCells();
```

Comment:
This could even move up to line 73, so that `row.getCells()` on line 74 can use `cells` instead